test/rubygems/test_gem_commands_sources_command.rb

```diff
@@ -107,6 +107,39 @@ def test_execute_add_allow_typo_squatting_source
     assert_empty ui.error
   end
 
+  def test_execute_add_allow_typo_squatting_source_forced
+    rubygems_org = "https://rubyems.org"
+
+    spec_fetcher do |fetcher|
+      fetcher.spec("a", 1)
+    end
+
+    specs = Gem::Specification.map do |spec|
+      [spec.name, spec.version, spec.original_platform]
+    end
+
+    specs_dump_gz = StringIO.new
+    Zlib::GzipWriter.wrap(specs_dump_gz) do |io|
+      Marshal.dump(specs, io)
+    end
+
+    @fetcher.data["#{rubygems_org}/specs.#{@marshal_version}.gz"] = specs_dump_gz.string
+    @cmd.handle_options %W[--force --add #{rubygems_org}]
+    ui = Gem::MockGemUi.new("y")
```

Comment:
If I understand this test correctly, you don't need a special UI here mocking input, since no questions will be asked.

Reply:
ah good point, yes I can remove that.

Reply:
Updated tests with the mock wrapper remove have been rebased into a new commit and pushed